plugins/jvm-abi-gen/src/org/jetbrains/kotlin/jvm/abi/JvmAbiClassBuilderInterceptor.kt

```diff
@@ -154,14 +155,16 @@ class JvmAbiClassBuilderInterceptor : ClassGeneratorExtension {
         override fun done(generateSmapCopyToAnnotation: Boolean) {
             // Remove local or anonymous classes unless they are in the scope of an inline function and
             // strip non-inline methods from all other classes.
-            when {
-                publicAbi ->
-                    abiClassInfo[internalName] = AbiClassInfo.Public
-                !localOrAnonymousClass && !isWhenMappingClass -> {
+            abiClassInfo[internalName] = when {
+                keepEverything -> AbiClassInfo.Public
+                classRemovedFromAbi -> AbiClassInfo.Delete
```

Comment:
```suggestion
                removeClassFromAbi -> AbiClassInfo.Delete
```

or similar naming would be more correct it seems